src/ddscxx/include/org/eclipse/cyclonedds/sub/AnyDataReaderDelegate.hpp

```diff
@@ -102,6 +102,14 @@ class OMG_DDS_API AnyDataReaderDelegate : public org::eclipse::cyclonedds::core:
                           const dds::topic::TopicDescription& td);
     virtual ~AnyDataReaderDelegate();
 
+
+    virtual void copy_samples(
+      dds::sub::detail::SamplesHolder& samples,
+      void**& c_sample_pointers,
+      dds_sample_info_t*& c_sample_infos,
+      int num_read) {
+      assert(1);
+    }
```

Comment:
Is it your intention to provide the implementation in the header file? I'm not too familiar with the C++ api, but this seems like an unnecessary change? @e-hndrks, can you shed some light perhaps?

Reply:
This function needed to be implemented to get the helloworld example with the templates to compile.